Storyboard struct -instantiate* methods now return class types qualified by module

nhojb opened the pull request:
This fixes issue #114, where instantiating a view controller with the same name as the storyboard
would resolve to the Storyboard struct type, not the module's view controller type.

nhojb committed:
Storyboard struct -instantiate* methods now return class types qualified with the customModule name (if present).

This fixes issue #114, where instantiating a view controller with the same name as the storyboard
would resolve to the Storyboard struct type, not the module's view controller type.

phimage commented:
Hi,
To test your modification, you can generate the Storyboards.swift files from the NatalieExample projects.
This will show if there is no compilation issue with simple case.

This must be tested with controller from swift framework also.

nhojb commented:
NatalieExample targets compile without error - Storyboard diffs are attached.

I'll add a test with a swift framework later.

[ExampleStoryboards.diff.txt](https://github.com/krzyzanowskim/Natalie/files/1678496/ExampleStoryboards.diff.txt)
[OSXExampleStorboards.diff.txt](https://github.com/krzyzanowskim/Natalie/files/1678497/OSXExampleStorboards.diff.txt)

krzyzanowskim commented:
It's ok to commit the changes to examples.

nhojb committed:
Fixes issue where the storyboardIdentifier property's access control was incorrect
when creating the IdentifiableProtocol extension for a view controller defined in an external module.

Was internal, should be public to match the access control of the external view controller.

nhojb commented:
Just created a new PR with the updated examples and inclusion of a framework view controller.
Includes the same (core) changes as this PR.

krzyzanowskim commented:
hm... close this one without merge then?
btw. you can update PR's by simply commit and push more.
